### PR TITLE
Add Recent Blogs to Home Page

### PR DESCRIPTION
**Pull request by anchalshivank**
- Updated blog plugin configuration.
- Disabled the default blog plugin.
- Implemented rendering of recent blogs on the home page (`index.jsx`).
- Recent blog data sourced from `recent-posts.json` in the `.docusaurus` directory.

### CODE REVIEW

File: plugins/blog-plugin.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        };
+      });
+
+      fs.mkdirSync(dir, {
+        recursive: true, // Avoid error if directory already exists.
+      });
+      // this is to update the recent-posts.json
+      // the same json that is imported in the index.jsx (Home page)
+      const fd = fs.openSync(`${dir}/recent-posts.json`, 'w');
+      fs.writeSync(fd, JSON.stringify(recentPosts));
```
Comment (sergefdrv) [inline plugins/blog-plugin.js]: I don't like this writing directly to the file system. Did you have a look at [this example](https://github.com/facebook/docusaurus/issues/4138#issuecomment-1703083513)? 

I also [asked](https://github.com/facebook/docusaurus/discussions/6423#discussioncomment-10235638) if there's a better way to do that in the most recent version of Docusaurus.

Comment (anchalshivank) [reply plugins/blog-plugin.js]: Understood, will look into this.

File: docusaurus.config.js [hunk 2]
```diff
@@ -57,15 +67,7 @@ const config = {
...
           editUrl: `${siteGitHubUrl}/edit/main/`,
         },
-        blog: {
-          blogTitle: 'Replica_IO Blog',
-          blogDescription: 'Blog of the Replica_IO project - an open-source framework for building practical distributed replication mechanisms.',
-          showReadingTime: true,
-          showLastUpdateTime: true,
-          showLastUpdateAuthor: true,
-          // Remove this to remove the "edit this page" links.
-          editUrl: `${siteGitHubUrl}/edit/main/`,
```
Comment (sergefdrv) [inline docusaurus.config.js:67]: Please preserve these properties.

File: docusaurus.config.js [hunk 3]
```diff
@@ -175,4 +177,4 @@ const config = {
     }),
 };
 
-export default config;
+export default config;
```
Comment (sergefdrv) [inline docusaurus.config.js:180]: Your formatter seems to be very aggressive: why you remove newlines at the end of files?

File: docusaurus.config.js (hunk 2 above)
Comment (anchalshivank) [reply docusaurus.config.js:67]: Ok

**Pull request closed without merging**